src/app/shared/preview-results/preview-results.component.ts

```diff
@@ -56,6 +56,7 @@ export class PreviewResultsComponent implements OnInit, OnDestroy {
   public activeIndex = 0;
 
   private allSubscriptions = new Subscription();
+  private dataSubscriptions = new Subscription();
```

Comment:
Where is `dataSubscriptions` except for unsubscribing this empty subscription?

Reply:
There is more serious problem to that. Currently, I believe there is a leaking subscription and this is prepared to solve the problem in the subsequent PR.